### PR TITLE
Fixes nested custom scalar

### LINKED ISSUE #29: bug: scalar custom nested subfield selections should be expanded
Query:
```
const result = client
    .query('ActionGoal')
    .getMany({
      select: {
        action: {
          select: {
            id: true,
            slug: true,
            photo: true,
            shareImage: true,
            title: true,
            titleObjectTemplate: true,
            url: true,
            description: true,
            discoveryHeader: true,
            discoveryDescription: true,
            notificationText: true,
            notificationObjectTemplate: true,
            enableNotifications: true,
            enableNotificationsText: true,
            search: true,
            location: true,
            locationRadius: true,
            timeRequired: true,
            startDate: true,
            endDate: true,
            approved: true,
            published: true,
            isPrivate: true,
            rewardAmount: true,
            activityFeedText: true,
            callToAction: true,
            completedActionText: true,
            alreadyCompletedActionText: true,
            selfVerifiable: true,
            isRecurring: true,
            recurringInterval: true,
            oncePerObject: true,
            minimumGroupMembers: true,
            limitedToLocation: true,
            tags: true,
            createdBy: true,
            updatedBy: true,
            createdAt: true,
            updatedAt: true,
            groupId: true,
            ownerId: true,
            objectTypeId: true,
            rewardId: true,
            verifyRewardId: true
          }
        },
        goal: {
          select: {
            id: true,
            name: true,
            slug: true,
            shortName: true,
            icon: true,
            subHead: true,
            tags: true,
            search: true,
            createdBy: true,
            updatedBy: true,
            createdAt: true,
            updatedAt: true
          }
        }
      }
    })
    .print();
```

Expected: output should further expand `location { x y }`, `timeRequired`...etc of custom types
Actual: `location` is just `location`, no sub fields

### PR DESCRIPTION
**Pull request by yyyyaaa**
fixes #29

### CODE REVIEW

**Comment by yyyyaaa** [comment]
cc @pyramation please help me review this

File: packages/introspectron/__tests__/gql.test.js [hunk 1]
```diff
@@ -20,6 +20,12 @@ it('mutationsDbe', () => {
   expect(mutationsDbe).toMatchSnapshot();
 });
 
+it.only('includes custom scalar types', () => {
```
Comment (pyramation) [inline packages/introspectron/__tests__/gql.test.js:23]: is the `.only` intentional here? I forget these too sometimes ;)

File: packages/introspectron/src/gql.js [hunk 2]
```diff
@@ -265,7 +265,7 @@ function parseSelectionObject(context, model, nesting) {
   throwIfInvalidContext(context);
 
   const selectionFields = HASH[model].fields.filter(
-    (f) => f.type.kind !== 'OBJECT'
+    (f) => !isPureObjectType(f.type)
```
Comment (pyramation) [inline packages/introspectron/src/gql.js:268]: super clean!

File: packages/launchql-client/__tests__/client.test.js [hunk 3]
```diff
@@ -24,7 +23,7 @@ describe('getMany', () => {
   it('should whitelist selected fields', () => {
     const client = new Client({
       meta: metaObject,
-      introspection: { ...queries, ...mutations }
+      introspection
```
Comment (pyramation) [inline packages/launchql-client/__tests__/client.test.js:26]: oh this is way better! so many nice refactors ;)

**Review by pyramation — APPROVED** [review]
LGTM!

I think maybe check on the test that has `only` to make sure the other tests are still ok, but otherwise looks amazing!

File: packages/introspectron/__tests__/gql.test.js (hunk 1 above)
Comment (yyyyaaa) [reply packages/introspectron/__tests__/gql.test.js:23]: yep, I forgot these, let me remove

**Pull request merged**